app/controllers/application_controller/advanced_search.rb

```diff
@@ -164,13 +164,21 @@ def adv_search_redraw_tree(tree)
       page << javascript_prologue
       tree_name = x_active_tree.to_s
       page.replace("#{tree_name}_div", :partial => "shared/tree", :locals => {:tree => tree, :name => tree_name})
+      # Redraw Advanced search for explorer screens (or storage, configuration scripts tree)
+      display_mode = params[:button] == 'save' ? params[:button] : nil
+      page.replace("adv_search_body",   :partial => "layouts/adv_search_body",   :locals => {:mode => display_mode})
+      page.replace("adv_search_footer", :partial => "layouts/adv_search_footer", :locals => {:mode => display_mode})
     end
   end
 
   def adv_search_redraw_listnav
     render :update do |page|
       page << javascript_prologue
       page.replace(:listnav_div, :partial => "layouts/listnav")
+      # Redraw Advanced search for non explorer screens
+      display_mode = params[:button] == 'save' ? params[:button] : nil
+      page.replace("adv_search_body",   :partial => "layouts/adv_search_body",   :locals => {:mode => display_mode})
+      page.replace("adv_search_footer", :partial => "layouts/adv_search_footer", :locals => {:mode => display_mode})
```

Comment:
Furthermore you are adding the same code 2x. That's one more reason to create a method. For it.

Then the new method can be called from the case blocks that need it not from these 2 methods that already do something different.

Thx!